### PR TITLE
Feature/aggregation p1 wrapup

### LINKED ISSUE #113: Message actions
- Display some actions in a bottom bar
- More actions in an "action sheet" (the list is contextual, see riot-android code)
- the "4 reactions" are part of #112 

https://app.zeplin.io/project/5c122fa790c5b4241ffa6be7/screen/5ca4dfa6305b0715a19e4caf
Screenshot 2019-04-29 at 15 47 12

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Stabilization PR for reaction phase 1

- [x]  Use emoji compatibility fallback for quick reactions, reactions pills, and reaction history
- [x]  Bug fixes on reactions (missing reactions, delay in local echo)
- [x]  Show simple reaction history (long click on existing reaction or via context menu)

image

- [x] Better support for big numbers in reactions
image

- [x] Use bigger font for short messages with only emojis
image

- [x] Added copy permalink action in menu

- [x] Added context menu for more items (membership, states)
image

Fixes issues with read receipt on invisible events (edits)
Fixes #113

### CODE REVIEW

File: vector/src/main/java/im/vector/riotredesign/features/home/room/detail/RoomDetailViewModel.kt [hunk 1]
```diff
@@ -55,7 +55,8 @@ class RoomDetailViewModel(initialState: RoomDetailViewState,
     private val roomId = initialState.roomId
     private val eventId = initialState.eventId
     private val displayedEventsObservable = BehaviorRelay.create<RoomDetailActions.EventDisplayed>()
-    private val timeline = room.createTimeline(eventId, TimelineDisplayableEvents.DISPLAYABLE_TYPES)
+    private val timeline = room.createTimeline(eventId,
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/features/home/room/detail/RoomDetailViewModel.kt]: Just split this in a variable please

File: vector/src/main/java/im/vector/riotredesign/features/home/room/detail/timeline/action/MessageActionsBottomSheet.kt [hunk 2]
```diff
@@ -117,36 +117,36 @@ class MessageActionsBottomSheet : BaseMvRxBottomSheetDialog() {
...
-                    .circleCrop()
-                    .placeholder(AvatarRenderer.getPlaceholderDrawable(requireContext(), it.userId, it.senderName))
-                    .into(senderAvatarImageView)
+        if (it.showPreview) {
+            bottom_sheet_message_preview.isVisible = true
+            senderNameTextView.text = it.senderName
+            messageBodyTextView.text = it.messageBody
+            messageTimestampText.text = it.ts
+
+            GlideApp.with(this).clear(senderAvatarImageView)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotredesign/features/home/room/detail/timeline/action/MessageActionsBottomSheet.kt]: Why not using AvatarRenderer for the whole stuff?

Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotredesign/features/home/room/detail/timeline/action/MessageActionsBottomSheet.kt]: yes, thx

File: vector/src/main/java/im/vector/riotredesign/features/home/room/detail/RoomDetailViewModel.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotredesign/features/home/room/detail/RoomDetailViewModel.kt]: done

**Pull request merged**